File: algo-character-count/python/char_count.py
def char_count(str):
    char_count_dict = {}
    for letter in str.lower():
        if letter.isalnum():
            char_count_dict[letter] = char_count_dict.get(letter, 0) + 1
    char_count_list = [[item, char_count_dict[item]] for item in char_count_dict]
    char_count_list.sort(key=lambda elem: (-elem[1], elem[0]))
    return char_count_list

File: algo-character-count/js/charCount.js
function compareElems(a, b) {
  if (a[1] > b[1]) {
    return -1;
  } else if (a[1] < b[1]) {
    return 1;
  }
  if (a[0] < b[0]) {
    return -1;
  } else if (a[0] > b[0]) {
    return 1;
  }
  return 0;
}
const charCount = function (string) {
  const charCountDict = {};
  const charCountArr = [];
  const alphanumeric = "abcdefghijklmnopqrstuvwxyz0123456789";
  for (let letter of string.toLowerCase()) {
    if (alphanumeric.indexOf(letter) !== -1) {
      charCountDict[letter] = charCountDict[letter] + 1 || 1;
    }
  }
  for (let item in charCountDict) {
    charCountArr.push([item, charCountDict[item]]);
  }
  charCountArr.sort(compareElems);
  return charCountArr;
};

module.exports = { charCount };
